libr/core/cmd_zign.c

```diff
@@ -80,6 +80,36 @@ static void cmd_zign_init(RCore *core) {
 	DEFINE_CMD_DESCRIPTOR (core, zs);
 }
 
+static bool addFcnHash(RCore *core, RAnalFunction *fcn, const char *name) {
+	bool retval = false;
+
+	if (!core || !fcn || !name) {
+		return retval;
+	}
+	
+	int maxsz = r_config_get_i (core->config, "zign.maxsz");
+	int fcnlen = r_anal_fcn_realsize (fcn);
+	int len = R_MIN (core->io->addrbytes * fcnlen, maxsz);
+
+	ut8 *buf = malloc (len);
+	if (!buf) {
+		return retval;
+	}
+
+	if (!r_io_is_valid_offset (core->io, fcn->addr, 0)) {
```

Comment:
you can check this before allocating the buffer and remove this unnecessary goto